modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/connection/ConnectionProviderObjectFactory.java

```diff
@@ -90,6 +90,45 @@ public ConnectionProviderResolver doGetObject() {
     return new ConnectionProviderResolver<>(builder, resolverSet, muleContext);
   }
 
+  private ConnectionProviderObjectBuilder resolveOAuthBuilder(ResolverSet resolverSet) {
+    OAuthGrantType grantType = providerModel.getModelProperty(OAuthModelProperty.class)
+        .map(OAuthModelProperty::getGrantTypes)
+        .get().get(0);
```

Comment:
get().get(0) seems super harsh, use ifPresent + forEach or fail if more thab 1

Reply:
the get(0) is because of a design error in 4.1 giving the property a List of grant types instead of just one. That list will never be empty and will never have more than one item